neuralpredictors/layers/readouts/multi_readout.py

```diff
@@ -86,37 +86,40 @@ class MultiReadoutSharedParametersBase(MultiReadoutBase):
     For more information on which parameters can be shared, refer for example to the FullGaussian2d readout
     """
 
-    def prepare_readout_kwargs(self, i, data_key, first_data_key, **kwargs):
+    def prepare_readout_kwargs(
+        self,
+        i,
+        data_key,
+        first_data_key,
+        share_transform=None,
```

Comment:
I think logic-wise, it only makes sense to let `share_transform`, `share_grid` and `share_features` default to boolean `False`, while `grid_mean_predictor` should default to `None`. But as explained in my main comment, `grid_mean_predictor` needs to be part of the `readout_kwargs` that ultimately get returned by the function

Reply:
I think it is cleaner to have them default to None, when they are not expected to be passed in to the function.

Reply:
Mh, I disagree... Their only purpose is this one function and in there, they have a clear meaning and task, which is boolean. What does it mean if someone passes in a list for `share_transform` and the list has no meaning but it is treated as a `True`? Whenever people look up what arguments they could give to their MultiReadout with shared parameters, this is the place where they will look and they should see what type these arguments are. This being said, we should also have a docstring there.

Reply:
I agree with @kklurz on this. `share_transform` etc. are bools and the default mode is that you do not share. If I would see `None` as a default that would suggest that 1) the default value is mutable, 2) I have to always specify whether I want to share or not and think about it, which seems like it's not the point that we are trying to make with this function.

Reply:
I think it depends on what kinds of values they are expected to have (i.e. what type they are expected to be). As far as I understand `share_transform`, `share_grid`, and `share_features` are expected to be boolean, and `grid_mean_predictor` can be a string. So my suggestion would be defaulting the booleans to some default boolean and the string to None - similar to what @kklurz mentioned.

Reply:
Ok thanks! I'm convinced here. I'll turn all but `grid_mean_predictor` into bools!

Reply:
I agree that boolean parameters should be `False` by default and the string one should be `None`. Also right now the `if` statement on line 109 is `True` if `share_transform` is `False`. Is that intended?

Reply:
Nevermind. I was looking at an older version...